src/game/Spell.cpp

```diff
@@ -5100,9 +5100,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                     else
                     {
                         Pet* dbPet = new Pet;
-                        if (dbPet->LoadPetFromDB((Player*)m_caster, 0))
-                            return SPELL_CAST_OK;           // still returns an error to the player, so this error must come from somewhere else...
-                        else
+                        if (!dbPet->TryLoadFromDB((Player*)m_caster, 0))
```

Comment:
that was poorly worded, you can of course have them go through summonPet, but if you use the function NewSummon->LoadPetFromDB((Player*)m_caster, 0) and the pet is dead then you're summoning the pet dead, which is not blizzlike, it shouldn't be allowed to be summoned dead, so if you don't plan to stop that in checkCast you must stop it before doing LoadPetFromDB